project/apps/web/app/collab/page.tsx

```diff
@@ -0,0 +1,50 @@
+'use client';
+
+import { HocuspocusProvider } from '@hocuspocus/provider';
+import Editor, { useMonaco } from '@monaco-editor/react';
+import { useEffect } from 'react';
+import { MonacoBinding } from 'y-monaco';
+import * as Y from 'yjs';
+
+import { env } from '@/env.mjs';
+
+export default function CollabPage() {
+  const monaco = useMonaco();
+
+  useEffect(() => {
+    if (typeof window !== 'undefined') {
+      if (monaco) {
+        const ydoc = new Y.Doc();
+
+        const provider = new HocuspocusProvider({
+          url: env.NEXT_PUBLIC_COLLAB_SOCKET_URL,
+          name: 'nextjs-monaco-demo',
```

Comment:
commenting to leave a note here:
`name` field will be provided by the backend, which will be unique for each collaboration room